zipline/_protocol.pyx

```diff
@@ -45,6 +45,11 @@ cdef class BarData:
         try:
             view = self._views[asset]
         except KeyError:
+            try:
+                asset = self.data_portal.env.asset_finder.retrieve_asset(asset)
```

Comment:
Right, but if `self._views[asset]` failed, doing a no-op on the asset isn't going to solve the problem, right?